File: src/OpenSilver.Showcase/Other/MarkupExtensions/ResxExtension.cs
using System;
using System.Windows.Markup;

namespace OpenSilver.Showcase;

[ContentProperty(nameof(Key))]
public class ResxExtension : MarkupExtension
{
    public string Key { get; set; }

    public override object ProvideValue(IServiceProvider serviceProvider)
    {
        return SampleResourceFile.ResourceManager.GetString(Key);
    }
}

File: src/OpenSilver.Showcase/Other/MarkupExtensions/ResxExtension.vb
Imports System.Windows.Markup

Namespace OpenSilver.Showcase

    <ContentProperty("Key")>
    Public Class ResxExtension
        Inherits MarkupExtension

        Public Property Key As String

        Public Overrides Function ProvideValue(serviceProvider As IServiceProvider) As Object
            Return SampleResourceFile.ResourceManager.GetString(Key)
        End Function
    End Class

End Namespace
